kythe/go/serving/explore/explore_test.go

```diff
@@ -0,0 +1,478 @@
+		p1c1: &srvpb.Relatives{
+			Tickets: []string{p1},
+			Type:    srvpb.Relatives_PARENTS,
+		},
+		p1c2: &srvpb.Relatives{
+			Tickets: []string{p2},
+			Type:    srvpb.Relatives_PARENTS,
+		},
+		badType: &srvpb.Relatives{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Relatives_CHILDREN,
+		},
+	}
+
+	functionToCallers = &protoTable{
+		f1: &srvpb.Callgraph{
+			Tickets: []string{f1r1, fr},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		f2: &srvpb.Callgraph{
+			Tickets: []string{f2r1, fr},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		f3: &srvpb.Callgraph{
+			Tickets: []string{f1, fr, f3},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		fr: &srvpb.Callgraph{
+			Tickets: []string{f2},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+		badType: &srvpb.Callgraph{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+	}
+
+	functionToCallees = &protoTable{
+		f1: &srvpb.Callgraph{
+			Tickets: []string{f3},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f2: &srvpb.Callgraph{
+			Tickets: []string{fr},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f3: &srvpb.Callgraph{
+			Tickets: []string{f3},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		fr: &srvpb.Callgraph{
+			Tickets: []string{f1, f2, f3},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f1r1: &srvpb.Callgraph{
+			Tickets: []string{f1},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		f2r1: &srvpb.Callgraph{
+			Tickets: []string{f2},
+			Type:    srvpb.Callgraph_CALLEE,
+		},
+		badType: &srvpb.Callgraph{
+			Tickets: []string{dontcare},
+			Type:    srvpb.Callgraph_CALLER,
+		},
+	}
+)
+
+func TestChildren_badData(t *testing.T) {
+	svc := Construct(t)
+
+	reply, err := svc.Children(ctx, &epb.ChildrenRequest{
+		Tickets: []string{badType},
+	})
+	if err == nil {
+		t.Errorf("Expected Children error for bad data, got: %v", reply)
+	}
+}
+
+func TestChildren_noData(t *testing.T) {
+	svc := Construct(t)
+
+	reply, err := svc.Children(ctx, &epb.ChildrenRequest{
+		Tickets: []string{"key with no data"},
+	})
+	if err != nil {
+		testutil.FatalOnErrT(t, "Children error: %v", err)
+	}
+	if len(reply.InputToChildren) != 0 {
+		t.Errorf("Expected empty response for missing key, got: %v", reply)
+	}
+}
+
+func TestChildren(t *testing.T) {
+	svc := Construct(t)
+	request := &epb.ChildrenRequest{
+		Tickets: []string{p1},
+	}
+
+	reply, err := svc.Children(ctx, request)
+	testutil.FatalOnErrT(t, "Children error: %v", err)
+
+	expectedInputToChildren := map[string]*epb.Tickets{
+		p1: {
+			Tickets: []string{p1c1, p1c2},
+		},
+	}
+
+	if !equalTicketsKeys(reply.InputToChildren, expectedInputToChildren) {
+		t.Errorf("Expected results and actual results have different ticket key sets:\n"+
+			"expected map: %v\n actual map: %v", expectedInputToChildren, reply.InputToChildren)
+	}
+
+	for ticket, expectedChildren := range expectedInputToChildren {
+		children := reply.InputToChildren[ticket]
+		if !reflect.DeepEqual(listToSet(expectedChildren.Tickets), listToSet(children.Tickets)) {
```

Comment:
I suggest using `github.com/google/go-cmp/cmp` to replace all of these `DeepEqual` checks.  As an added benefit, you can use the `cmp.Transformer` option to sort the elements before comparing and it'll remove the boilerplate of
```
if !reflect.DeepEqual(f(x), f(y)) {
  t.Errorf("..............................................", x, y)
}
```

 to just
```
if diff := cmp.Diff(x, y, fTransformer); diff != "" {
  t.Errorf("Found different ____ (-: found; +: expected)\n%s", diff)
}
```